### DISCUSSION TITLE
Using .net8-android and MAUI packages in the same project

### DISCUSSION THREAD

**Discussion by ThysBrits** [dotnet/maui#20807] [Upgrading from Xamarin]
We are making the move from an older Xamarin Android project to the newer project template, using VS2022. We have a few issues with some of our NuGet packages that have been giving us issues. We tried the MAUI upgrade tool route, but that was a disaster, nothing wanted to build. Then we created a new blank Android Application project and started adding our existing XAML and other class files to the project, starting with the main screen and adding bit by bit when we have a running app, making small changes where required. The biggest issue is finding and matching NuGet packages. We now have multiple AndroidX NuGet packages added to the project as well, which are required for some of the (very simple) forms. My question is - how dangerous is it mixing the two seemingly different technologies? 

Also, our app makes use of ZXing.Net.Mobile, but neither ZXing.Net.Mobile or ZXing.Net.MAUI work with the newer app, as it brings incompatible other packages that give horrible JAVA build errors.

Similar is the Raygun NuGet package, of which neither the MAUI version or the Mobile one gives us a buildable project.

Where can we find support for these kinds of issues? It appears as though we are trying to work with bleeding-edge technology, but it's already 4 years old.

**Comment by Geera612** [comment]
Did you find a solution. I have the same problem.

**Reply by ThysBrits** [reply]
We hope so. We have created a new MAUI app instead, and this required changing a lot of the front-end pages to using the newer components, but it means that everything is now MAUI and AndroidX, so we can actually use the updated ZXing.Net.MAUI component too. It's still in progress, but we are hopeful.

**Comment by davidortinau** [comment]
@ThysBrits when you run into build errors like you describe, please share the errors so others can comment on it. There's not enough information in your initial post.

> how dangerous is it mixing the two seemingly different technologies?

What two technologies are you referring to? A .NET MAUI app uses .NET for Android which uses AndroidX. That's all related. 

The main difference to avoid is .NET vs .NET Framework (Xamarin) when it comes to dependencies. Look at the target frameworks on NuGet.org.

Was your Xamarin Android project using Xamarin.Forms?
